.github/workflows/ci-cleanup.yml

```diff
@@ -5,8 +5,29 @@ on:
       - closed
 
 jobs:
+  cleanup-mock:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Extract Branch Name
+        id: extract-branch
+        run: |
+          echo SUBDOMAIN=mock-$(echo ${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}} | perl -pe 's/[^a-zA-Z0-9]+/-/g and s/-+$//g' | tr '[:upper:]' '[:lower:]') >> $GITHUB_OUTPUT
+      - name: Check If DNS Record Exists
+        id: cloudflare-dns-check
+        uses: fjogeleit/http-request-action@v1
+        with:
+          url: 'https://api.cloudflare.com/client/v4/zones/${{ secrets.CF_ZONE_ID }}/dns_records?name=${{steps.extract-branch.outputs.SUBDOMAIN}}.${{secrets.FPJS_CI_DOMAIN}}&tags=owner:akamai-integration-ci'
+          method: 'GET'
+          customHeaders: '{"Authorization": "Bearer ${{ secrets.CF_AUTH_TOKEN }}"}'
+      - name: Delete Cloudflare DNS Record
+        if: ${{ fromJson(steps.cloudflare-dns-check.outputs.response).result_info.total_count > 0 }}
+        uses: fjogeleit/http-request-action@v1
+        with:
+          url: 'https://api.cloudflare.com/client/v4/zones/${{ secrets.CF_ZONE_ID }}/dns_records/${{ fromJson(steps.cloudflare-dns-check.outputs.response).result[0].id }}'
+          method: 'DELETE'
+          customHeaders: '{"Authorization": "Bearer ${{ secrets.CF_AUTH_TOKEN }}"}'
+
   cleanup:
-    if: github.event.pull_request.merged == true
```

Comment:
I believe this is a mistake. Could you undo this, and also add the same line to the `cleanup-mock` job? @TheUnderScorer

Reply:
Doesn't it make sense to have cleanup whenever PR is closed, not only merged?

Reply:
I'll revert this for now, so that we can close this PR. Later we can decide if it makes sense to keep this change or not. WDYT?